src/Agent.Worker/JobExtension.cs

```diff
@@ -118,6 +118,31 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
 
                         taskConditionMap[task.Id] = condition;
                     }
+                    foreach (var knob in AgentKnobs.GetAllKnobs())
+                    {
+                        var value = knob.GetValue(jobContext);
+                        if (value.Source != KnobSource.BuiltInDefault)
+                        {
+                            var tag = "";
+                            if (knob.IsDeprecated)
+                            {
+                                tag = "(DEPRECATED)";
+                            }
+                            else if (knob.IsExperimental)
+                            {
+                                tag = "(EXPERIMENTAL)";
+                            }
+                            var outputLine = $"{knob.Name}={value.AsString()} {value.Source} {value.Which} {tag}";
+                            if (!string.IsNullOrEmpty(tag))
```

Comment:
Does it make sense to warn on experimental features? That seems like a pretty explicit choice

Reply:
I made it a warning so that it would stand out to someone doing debug. I'm fine if we only want to warn on deprecated knobs

Reply:
I'd lean that direction mostly just because there seems to be a sizable number of people who aren't ok with warnings showing up ever. With that said, I don't feel passionately about it and am fine leaving it

Reply:
One other option: we could warn by default but create a knob to change that behavior :)